v8_c_api/src/v8_c_api.cpp

```diff
@@ -1299,7 +1299,19 @@ void v8_ObjectSetInternalField(v8_local_object *obj, size_t index, v8_local_valu
 }
 
 v8_local_value* v8_ObjectGetInternalField(v8_local_object *obj, size_t index) {
+    // On v8 version 11.9, the API was changed to allow setting any V8 data and not just
+    // Local<Value>. This is why we need to check the V8 version to decide which code
+    // flow to take.
+#if (V8_MAJOR_VERSION > 11) || (V8_MAJOR_VERSION == 11 && V8_MINOR_VERSION >= 9)
+    v8::Local<v8::Data> data = obj->obj->GetInternalField(index);
+    if (!data->IsValue())
+    {
+        return NULL;
+    }
+    v8::Local<v8::Value> val = data.As<v8::Value>();
+#else
     v8::Local<v8::Value> val = obj->obj->GetInternalField(index);
+#endif
```

Comment:
The change is that `GetInternalField` now returns Local<Data> and before it returned Local<Value>. So now the static assert will not work and we need a runtime check.

Reply:
@vityafx please look at the `main` branch of V8. As I already wrote the API was change to return `Local<Data>` instead of `Local<Value>`. This is why we need to check if the returned `Local<Data>` is in fact `Local<Value>` and if it does we need to cast it.
This is exactly what the PR does if the version has this new API with the change. Otherwise it keeps the code as is.

Reply:
We can use it without the `#ifdef` but then we will not have the type checking and if the `Local<Data>` is not `Local<Value>` it will crash.

Reply:
So first its a test and you do not mind it will crash.
Second when I came with this argument here: https://github.com/RedisGears/v8-rs/pull/51#discussion_r1314186919
Where it is impossible to have an isolate without an ID you were against asserting it. Why here your suggestion is the other way around.
My general opinion is that we should assert things that can not happened (even if the V8 api suggest otherwise), but I believe it is more important to be consistent, this is why I chose to check it.

Reply:
@vityafx the goal of this PR is to fix the build issue. We can think about extending the API but this should be done on another PR. I see now that our rust code can not handle NULL return value so I suggest to just us the `As<v8::Value>()` anyway and it will assert if it fails. We can change the API later on..

Reply:
Issue opened: https://github.com/RedisGears/v8-rs/issues/59